src/components/mdpDatePicker/mdpDatePicker.js

```diff
@@ -461,6 +461,54 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
                 scope.$on("$destroy", function() {
                     inputElement.off("reset input blur", onInputElementEvents);
                 });
+
+                if(scope.dateFormat == "MM/DD/YYYY") {
+                    inputElement.on("keypress", formatDateAsTyped);
+                }
+
+
+                function formatDateAsTyped(event) {
+                    var inputDate = this.value;
+                    var unicode = event.keyCode ? event.keyCode : event.charCode;
+
+                    //Allow BackSpace, Del, left arrow, right arrow in firefox. IE, chrome, safari supports these keys
+                    if (event.keyCode == 8 || event.keyCode == 46 || (event.keyCode == 37 && !event.charCode) || event.keyCode == 39) {
+                        return true;
+                    }
+
+                    if (isNumber(unicode) && inputDate.length < 10) {
+                            //replaces 2 to 02, 3 to 03.. for month field..
+                            inputDate = inputDate.replace(/^([2-9])/, "0$1");
+                            //restrict typing 13, 14, 15, 16, 17, 18, 19 for month field
+                            if(inputDate == "1" && (unicode >= 51 && unicode <= 57)) {
+                            return false;
+                            }
+                            //replaces 4 to 04, 5 to 05.. for date field..
+                            inputDate = inputDate.replace(/^([0-9])([0-9])(\/)([4-9])/, "$1$2$30$4");
```

Comment:
Please double-check that this works; I'm not sure about JavaScript's implementation of back-references, but this could be interpreted as "one,two,thirty,four"